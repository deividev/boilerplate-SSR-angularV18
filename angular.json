{
  "$schema": "./node_modules/@angular/cli/lib/config/schema.json",
  "version": 1,
  "newProjectRoot": "projects",
  "projects": {
    "Angular18": {
      "projectType": "application",
      "schematics": {
        "@schematics/angular:component": {
          "style": "scss",
          "changeDetection": "OnPush"
        }
      },
      "root": "",
      "sourceRoot": "src",
      "prefix": "app",
      "architect": {
        "build": {
          "builder": "@angular-devkit/build-angular:application",
          "options": {
            "outputPath": "dist/angular18",
            "index": "src/index.html",
            "browser": "src/main.ts",
            "polyfills": [],
            "tsConfig": "tsconfig.app.json",
            "inlineStyleLanguage": "scss",
            "assets": [
              {
                "glob": "**/*",
                "input": "public"
              }
            ],
            "styles": [
              "src/styles.scss",
              "node_modules/primeng/resources/themes/lara-light-blue/theme.css",
              "node_modules/primeng/resources/primeng.min.css"
            ],
            "scripts": [],
            "server": "src/main.server.ts",
            "prerender": true,
            "ssr": {
              "entry": "server.ts"
            }
          },
          "configurations": {
            "production": {
              "budgets": [
                {
                  "type": "initial",
                  "maximumWarning": "500kB",
                  "maximumError": "1MB"
                },
                {
                  "type": "anyComponentStyle",
                  "maximumWarning": "2kB",
                  "maximumError": "4kB"
                }
              ],
              "outputHashing": "all"
            },
            "development": {
              "optimization": false,
              "extractLicenses": false,
              "sourceMap": true
            }
          },
          "defaultConfiguration": "production"
        },
        "serve": {
          "builder": "@angular-devkit/build-angular:dev-server",
          "configurations": {
            "production": {
              "buildTarget": "Angular18:build:production"
            },
            "development": {
              "buildTarget": "Angular18:build:development"
            }
          },
          "defaultConfiguration": "development"
        },
        "extract-i18n": {
          "builder": "@angular-devkit/build-angular:extract-i18n"
        },
        "test": {
          "builder": "@angular-devkit/build-angular:karma",
          "options": {
            "config": "jest.config.js",
            "polyfills": [],
            "tsConfig": "tsconfig.spec.json",
            "inlineStyleLanguage": "scss",
            "assets": [
              {
                "glob": "**/*",
                "input": "public"
              },
              {
                "glob": "**/*",
                "input": "./",
                "output": "/"
              }
            ],
            "styles": [
              "src/styles.scss",
              "node_modules/primeng/resources/themes/lara-light-blue/theme.css",
              "node_modules/primeng/resources/primeng.min.css",
              {
                "input": "src/lara-dark-blue.scss",
                "bundleName": "lara-dark-blue",
                "inject": false
              },
              {
                "input": "src/lara-light-blue.scss",
                "bundleName": "lara-light-blue",
                "inject": false
              }
            ],
            "scripts": []
          }
        },
        "lint": {
          "builder": "@angular-eslint/builder:lint",
          "options": {
            "lintFilePatterns": [
              "src/**/*.ts",
              "src/**/*.html"
            ]
          }
        }
      }
    }
  },
  "cli": {
    "analytics": false,
    "schematicCollections": [
      "@angular-eslint/schematics"
    ]
  }
}
