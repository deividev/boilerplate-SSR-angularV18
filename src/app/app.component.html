<router-outlet />

<select (change)="changeTheme($event)">
  <option value="lara-light-blue">Lara Light Blue</option>
  <option value="lara-dark-blue">Lara Dark Blue</option>
</select>
<div class="card flex flex-column md:justify-content-between row-gap-3">
  <p-button label="Button 1"></p-button>
  <p-button styleClass="p-button-secondary" label="Button 2"></p-button>
  <p-button styleClass="p-button-help" label="Button 3"></p-button>
</div>
<app-main-container />
