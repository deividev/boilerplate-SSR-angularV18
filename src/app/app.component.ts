import { Component } from '@angular/core';
import { RouterOutlet } from '@angular/router';
import { MainContainerComponent } from '../components';
import { ThemeService } from '../services/theme.service';
import { ButtonModule } from 'primeng/button';

@Component({
  selector: 'app-root',
  standalone: true,
  imports: [RouterOutlet, MainContainerComponent, ButtonModule],
  templateUrl: './app.component.html',
  styleUrl: './app.component.scss',
})
export class AppComponent {
  title = 'Angular18-boilerplate';

  constructor(private themeService: ThemeService) {}

  changeTheme(event: Event): void {
    const theme = (event.target as HTMLSelectElement).value;
    this.themeService.setTheme(theme);
  }
}
