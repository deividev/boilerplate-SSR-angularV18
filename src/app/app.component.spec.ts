import { TestBed } from '@angular/core/testing';
import { AppComponent } from './app.component';
import {
  provideHttpClient,
  withInterceptorsFromDi,
} from '@angular/common/http';
import { render } from '@testing-library/angular';

describe('AppComponent', () => {
  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [AppComponent],
      providers: [provideHttpClient(withInterceptorsFromDi())],
    }).compileComponents();
  });

  it('should create the app', () => {
    const fixture = TestBed.createComponent(AppComponent);
    const app = fixture.componentInstance;
    expect(app).toBeTruthy();
  });

  it(`should have the 'Angular18-boilerplate' title`, () => {
    const fixture = TestBed.createComponent(AppComponent);
    const app = fixture.componentInstance;
    expect(app.title).toEqual('Angular18-boilerplate');
  });

  it('should match the snapshot', async () => {
    const { container } = await render(AppComponent);
    expect(container).toMatchSnapshot();
  });
});
