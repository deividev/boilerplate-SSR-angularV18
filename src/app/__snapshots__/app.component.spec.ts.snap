// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent should match the snapshot 1`] = `
<div
  id="root2"
>
  <router-outlet />
  
  <select>
    <option
      value="lara-light-blue"
    >
      Lara Light Blue
    </option>
    <option
      value="lara-dark-blue"
    >
      Lara Dark Blue
    </option>
  </select>
  <div
    class="card flex flex-column md:justify-content-between row-gap-3"
  >
    <p-button
      class="p-element"
      label="Button 1"
    >
      <button
        class="p-ripple p-element p-button p-component"
        data-pc-name="button"
        data-pc-section="root"
        pautofocus=""
        pripple=""
        type="button"
      >
        
        
        
        
        
        
        <span
          class="p-button-label"
          data-pc-section="label"
        >
          Button 1
        </span>
        
        
      </button>
    </p-button>
    <p-button
      class="p-element"
      label="Button 2"
      styleclass="p-button-secondary"
    >
      <button
        class="p-ripple p-element p-button p-component p-button-secondary"
        data-pc-name="button"
        data-pc-section="root"
        pautofocus=""
        pripple=""
        type="button"
      >
        
        
        
        
        
        
        <span
          class="p-button-label"
          data-pc-section="label"
        >
          Button 2
        </span>
        
        
      </button>
    </p-button>
    <p-button
      class="p-element"
      label="Button 3"
      styleclass="p-button-help"
    >
      <button
        class="p-ripple p-element p-button p-component p-button-help"
        data-pc-name="button"
        data-pc-section="root"
        pautofocus=""
        pripple=""
        type="button"
      >
        
        
        
        
        
        
        <span
          class="p-button-label"
          data-pc-section="label"
        >
          Button 3
        </span>
        
        
      </button>
    </p-button>
  </div>
  <app-main-container>
    <div>
      
    </div>
  </app-main-container>
</div>
`;
